Makefile

```diff
@@ -115,6 +115,10 @@ test-debug:
 	@echo Running tests in debug mode, excluding third party tests under vendor
 	go test -v $(shell go list ./... | grep -v -e /vendor/)
 
+test-race:
```

Comment:
I guess it could be good to put the test-race as part of the test target in the Makefile, just to not make it optional.
Or to add it in travis, too.

Reply:
Yep, I can do it as part of this PR if you wish.

Reply:
This is done in the last version of this PR.